tests/CMakeLists.txt

```diff
@@ -43,6 +43,14 @@ add_executable(bpftrace_test
   ${CODEGEN_SRC}
 )
 
+# Regenerate DWARF data for unit tests
+add_custom_target(testing_debuginfo_data
+        COMMAND
+        ${CMAKE_COMMAND}
+        -DDATA_SOURCE_DIR="${CMAKE_SOURCE_DIR}/tests/data"
+        -P ${CMAKE_SOURCE_DIR}/tests/data/generate_debuginfo_data.cmake)
```

Comment:
I'm not sure this is correct. Will this regen the final output if the `data_source.c` is changed?

Instead, how about using [add_custom_command()](https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command) with a shell script? The output file is the generated header. The input files are the shell script and `data_source.c`. And then the following line `add_dependencies` stays the same and I think it will "just work". Should be a littler easier to grok/program too.

Reply:
> I'm not sure this is correct. Will this regen the final output if the `data_source.c` is changed?

Yes, it will. CMake will handle that.

> Instead, how about using [add_custom_command()](https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command) with a shell script? The output file is the generated header. The input files are the shell script and `data_source.c`. And then the following line `add_dependencies` stays the same and I think it will "just work". Should be a littler easier to grok/program too.

In fact, using add_custom_command() was my original approach, but I wasn't able to make it work. I just couldn't make CMake create the correct rule for the target file.

Using a shell script instead of the CMake script is possible, but CMake allows to use configure_file() which gives quite a nice control of the contents of the generated header.

Reply:
Any chance this example might help? https://github.com/libbpf/libbpf-bootstrap/blob/67a29e511cc9d0a570d4d3b9797827f3a08ccdb5/tools/cmake/FindBpfObject.cmake#L163-L170

I'm ok to merge this as is too

Reply:
The example suggests to use target_sources() to create the dependency between the target (`bpftrace_test`) and the generated header, but unfortunately this doesn't rebuild `bpftrace_test` when `data_source.c` is changed.

It's probably possible to make it work somehow, but likely it will be more complicated than the current approach.